src/js/relayclient/RelayClient.js

```diff
@@ -412,8 +412,9 @@ class RelayClient {
     fixTransactionReceiptResp(resp) {
         if (resp && resp.result && resp.result.logs) {
             let logs = abi_decoder.decodeLogs(resp.result.logs);
-            let relayed = logs.find(e => e && e.name == 'TransactionRelayed');
-            if (relayed && relayed.events.find(e => e.name == "status").value != 0) {
+            let relayCallFailed = logs.find(e => e && e.name == 'RelayCallFailed');
+            let transactionRelayed = logs.find(e => e && e.name == 'TransactionRelayed');
+            if (relayCallFailed || (transactionRelayed && transactionRelayed.events.find(e => e.name == "status").value != 0)) {
                 console.log("reverted relayed transaction. changing status to zero");
```

Comment:
In case of failure, please report (log) the reason code (from either event..) - as its the only source of that information for a dapp creator (the error also appear in the relay log, but that's accessible only on local relay)